pallets/registrar/src/lib.rs

```diff
@@ -320,6 +344,17 @@ pub mod pallet {
 
     #[pallet::hooks]
     impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
+            // Account for on_finalize weight
+            let mut weight = Weight::zero().saturating_add(T::DbWeight::get().reads(1));
```

Comment:
isnt take a write?

Reply:
From what I understand, as the `BufferedParasToDeregister` storage is set inside `on_initialize` and then drained inside `on_finalize`, we don't need to account for an extra write, but maybe @tmpolaczyk can confirm.

Reply:
`take` is a write, but if I'm correct storage items that only exist inside a block doesn't count as writes because they don't affect the state root and they don't need PoV. Would be nice if substrate had some explicit "transactional" storage, but I think it doesn't. 

So the current implementation looks good to me.